adapters/avail/src/spec/transaction.rs

```diff
@@ -0,0 +1,76 @@
+#[cfg(feature = "native")]
+use avail_subxt::{
+    api::runtime_types::{da_control::pallet::Call, da_runtime::RuntimeCall::DataAvailability},
+    primitives::AppUncheckedExtrinsic,
+};
+use bytes::Bytes;
+#[cfg(feature = "native")]
+use codec::Encode;
+use serde::{Deserialize, Serialize};
+use sov_rollup_interface::da::{BlobReaderTrait, CountedBufReader};
+
+use super::address::AvailAddress;
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Debug)]
+
+pub struct AvailBlobTransaction {
+    blob: CountedBufReader<Bytes>,
+    hash: [u8; 32],
+    address: AvailAddress,
+}
+
+impl BlobReaderTrait for AvailBlobTransaction {
+    type Address = AvailAddress;
+
+    fn sender(&self) -> AvailAddress {
+        self.address.clone()
+    }
+
+    fn hash(&self) -> [u8; 32] {
+        self.hash
+    }
+
+    fn verified_data(&self) -> &[u8] {
+        self.blob.accumulator()
+    }
+
+    #[cfg(feature = "native")]
+    fn advance(&mut self, num_bytes: usize) -> &[u8] {
+        self.blob.advance(num_bytes);
+        self.verified_data()
+    }
+
+    fn total_len(&self) -> usize {
+        self.blob.total_len()
+    }
+}
+
+impl AvailBlobTransaction {
+    #[cfg(feature = "native")]
+    pub fn new(unchecked_extrinsic: &AppUncheckedExtrinsic) -> Self {
+        let address = match &unchecked_extrinsic.signature {
+            Some((subxt::utils::MultiAddress::Id(id), _, _)) => AvailAddress(id.clone().0),
+            _ => unimplemented!(),
```

Comment:
Is it possible to write a reason why this is not implemented like: `unimplemented("Not supported ...")`

Reply:
Done. Thanks @vibhurajeev